pkg/cluster/cluster.go

```diff
@@ -836,7 +841,7 @@ func (c *Cluster) compareServices(old, new *v1.Service) (bool, string) {
 	return true, ""
 }
 
-func (c *Cluster) compareLogicalBackupJob(cur, new *batchv1.CronJob) (match bool, reason string) {
+func (c *Cluster) compareLogicalBackupJob(cur, new *batchv1.CronJob, deletedPodAnnotations *[]string) (match bool, reason string) {
```

Comment:
could you create a result struct similar to the one we have for comparing stateful sets? For compareAnnotations I'm fine to keep it as is, but for compareLogicalBackupJob it should not require to many changes.